### ISSUE TITLE
Bandcamp connector

### ISSUE THREAD

**Issue by Kerkerseele** [web-scrobbler/web-scrobbler#4073]
![Untitled](https://github.com/web-scrobbler/web-scrobbler/assets/126416078/bb2d35e6-34c3-4522-9fc4-a1ed97024d5f)

The "only when clicked" setting for Bandcamp connector never gets saved because each artist page has a different url. I set the connector to "always allow" on bandcamp.com but I still need to click when listening to pretty much anything.

**Comment by yayuyokitano** [comment]
This is not something that we can do anything about, those options are from the browser side and we cannot change them.
Unfortunately, you will need to set it to always allow on every website to avoid this.

**Comment by Kerkerseele** [comment]
I can't seem to find that in the settings. An "always allow on every website" option would be cool.

**Comment by yayuyokitano** [comment]
Again these are not options we have any control over, it is your browser.
I think every browser has some way of enabling on every website, but its different for several of them, so for information, which browser are you using?

**Comment by Kerkerseele** [comment]
I use Firefox. 
![image](https://github.com/web-scrobbler/web-scrobbler/assets/126416078/b28d4f0d-3f29-4876-a636-ba929e91d521)

Also I'm confused. Aren't these options in the screenshot belonging to the extension itself? How do you not have control over them?

**Comment by Kerkerseele** [comment]
![image](https://github.com/web-scrobbler/web-scrobbler/assets/126416078/74510dbf-d962-45a1-bb19-b590807149b1)

Would one of these do the trick?

**Comment by yayuyokitano** [comment]
Yes, tick the "access your data for all websites" and the issue should resolve (you may need to refresh the bandcamp page for the setting to apply).

For information, while the "only when clicked" etc. options relate to how web scrobbler is allowed to work, firefox determines them, and you'll find every extension that accesses websites have the exact same options because of it.
It's basically a security barrier from firefox's side (because while we need to access that level of data for web scrobbler to work, hypothetically the same level of access allows malicious data scrubbing etc).
Only the options above that line in the screenshot, "disable connector" and "disable until tab is closed" are options that we control and that directly interact with our code.

**Comment by Kerkerseele** [comment]
It worked. Tysm

**Comment by yayuyokitano** [comment]
Glad to hear it!
